typescript/infra/src/xerc20/utils.ts

```diff
@@ -376,35 +442,80 @@ function getTxCallData(transactions: PopulatedTransaction[]): CallData[] {
 async function sendTransactions(
   multiProvider: MultiProvider,
   chain: string,
-  owner: Address,
   transactions: PopulatedTransaction[],
+  xERC20Address: Address,
   bridgeAddress: Address,
 ): Promise<void> {
+  // function aims to successfully submit a transaction for the following scenarios:
+  // 1. (initial deployment before ownership transfer) xERC20 is owned by an EOA (deployer), the expected owner (safe) is NOT the actual owner (deployer), the configured signer (deployer) is the owner (deployer) -> send normal transaction
+  // 2. xERC20 is owned by an EOA (deployer), the expected owner (deployer) is the actual owner (deployer) and the configured signer (deployer) is the owner (deployer) -> send normal transaction
+  // 3. xERC20 is owned by a Safe, the expected owner (safe) is the actual owner (safe), the configured signer (deployer) has the ability to propose safe transactions -> propose a safe transaction
+
   const signer = multiProvider.getSigner(chain);
-  const proposerAddress = await signer.getAddress();
-  const isSafeOwner = await checkSafeOwner(
-    proposerAddress,
+  const signerAddress = await signer.getAddress();
+  const ownable = Ownable__factory.connect(xERC20Address, signer);
+  const actualOwner = await ownable.owner();
+
+  // only attempt to send as safe if
+  // (a) the actual owner is a safe
+  // (b) the signer (deployer) has the ability to propose transactions on the safe
+  // otherwise fallback to a signer transaction, this fallback will allow for us to handle scenario 1 even though the expected owner is a safe
+  const isOwnerSafe = await checkOwnerIsSafe(
+    signerAddress,
     chain,
     multiProvider,
-    owner,
+    actualOwner,
+    bridgeAddress,
   );
 
-  if (isSafeOwner) {
-    await sendAsSafeMultiSend(
+  if (isOwnerSafe) {
+    const isSafeProposer = await checkSafeProposer(
```

Comment:
Probably should keep for a followup PR, but this seems rather ineffient, as we will call `getSafe` twice